exercise/core/serializers.py

```diff
@@ -0,0 +1,54 @@
+from rest_framework import serializers
+from .models import Recipe, Ingredient
+
+
+class IngredientSerializer(serializers.ModelSerializer):
+    """Serializer for ingredient model"""
+
+    class Meta:
+        model = Ingredient
+        fields = [
+            "name",
+        ]
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    """Serializer for recipe model"""
+
+    ingredients = IngredientSerializer(many=True, required=False)
+
+    class Meta:
+        model = Recipe
+        fields = ["id", "name", "description", "ingredients"]
+        read_only_fields = ("id",)
+
+    def create(self, validated_data):
+        ingredients = validated_data.pop("ingredients", None)
+        recipe = Recipe.objects.create(**validated_data)
+
+        if ingredients is not None:
+            for ingredient in ingredients:
```

Comment:
If you use an empty list as a default you can skip the `if`
```suggestion
        ingredients = validated_data.pop("ingredients", [])
        recipe = Recipe.objects.create(**validated_data)

        for ingredient in ingredients:
```